src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTagger.cs

```diff
@@ -19,56 +23,104 @@ internal sealed class InlineParameterNameHintsTagger : ITagger<IntraTextAdornmen
     {
         private readonly ITagAggregator<InlineParameterNameHintDataTag> _tagAggregator;
         private readonly ITextBuffer _buffer;
+        private readonly ITextView _textView;
+
+        /// <summary>
+        /// _cache stores the parameter hint tags in a global location 
+        /// </summary>
+        private readonly List<ITagSpan<IntraTextAdornmentTag>> _cache;
```

Comment:
i'm still opposed to teh cache.

Reply:
regardless of the performance, I do think I will still need some sort of logic where the tagspans get stored globally because when the options change, I still need to clear that out and completely recreate them

Reply:
i would say you *don't* need to do that.  intead, the way tagging works is you hook up "Event sources", i.e. things you're listening to taht should cause you to recompute tags.  in your case, you'd have an event source listening for wahtever option this is to change, and then firing the appropraite "something changed" notification to retrigger tagging.

Reply:
See ITaggerEventSource and TaggerEventSources.  We'll likely need a new source here, but it should hopefully not be hard to hook up and make work.

Reply:
I see, okay. I will look into it